src/hydro_forces.cpp

```diff
@@ -353,58 +353,48 @@ double TestHydro::setVelHistory(double val, int step, int b_num, int index) {
 *******************************************************************************/
 std::vector<double> TestHydro::ComputeForceHydrostatics() {
 	std::vector<double> position, displacement;
-	unsigned total_dofs = 6 * num_bodies;
-	position.resize(total_dofs, 0.0);
-	displacement.resize(total_dofs, 0.0);
+	assert(num_bodies > 0);
 
-	// orientation initialized to system current pos/rot vectors
 	for (int b = 0; b < num_bodies; b++) {
-		for (int i = 0; i < 3; i++) {
-			unsigned b_offset = 6 * b;
-			if (b_offset + i + 3 > total_dofs || b_offset + i < 0) {
-				std::cout << "temp index in hydrostatic force is bad " << std::endl;
-			}
-			position[i + b_offset] = bodies[b]->GetPos()[i];
-			position[i + 3 + b_offset] = bodies[b]->GetRot().Q_to_Euler123()[i]; // TODO check if we want this, or GetWvel_par()?
+		// initialize variables
+		auto body = bodies[b];
+		auto body_h5file = file_info[b];
+		auto rho = body_h5file.rho;
+		unsigned b_offset = 6 * b;  // force_hydrostatic has 6 elements for each body so to skip to the next body we move 6 spaces
+		auto body_force_hydrostatic = &force_hydrostatic[b_offset];
+		auto gg = body->GetSystem()->Get_G_acc().Length();
+
+		// hydrostatic stiffness due to offset from equilibrium
+		auto body_position = body->GetPos();
+		auto body_rotation = body->GetRot().Q_to_Euler123();
+		// calculate displacement
+		chrono::ChVectorN<double, 6> displacement;
+		for (int ii = 0; ii < 3; ii++) {
+			displacement[ii] = body_position[ii] - equilibrium[ii];
+			displacement[ii + 3] = body_rotation[ii] - equilibrium[ii + 3];
 		}
-	}
-
-	// make displacement vector for system
-	for (int i = 0; i < total_dofs; i++) {
-		displacement[i] = equilibrium[i] - position[i];
-	}
-
-	// re invent matrix vector multiplication
-	for (int b = 0; b < num_bodies; b++) {
-		unsigned b_offset = 6 * b;
-		for (int i = 0; i < 6; i++) {
-			for (int j = 0; j < 6; j++) {
-				double t = (file_info[b].GetHydrostaticStiffness(i, j));
-				//if (i == 4 && j == 4 && t < 0) {
-				//	t = -t;
-				//}
-				force_hydrostatic[i + b_offset] += t * displacement[j + b_offset];
-			}
+		// calculate force
+		auto force_offset = -gg * rho * body_h5file.lin_matrix * displacement;
```

Comment:
Don't use auto, unclear

Reply:
left it as is for now (auto) because of more complex type (for maintainability, see comment above)

Reply:
declared explicitly (see discussion below)